src/components/document/SubTabs/components/Meta/Meta.tsx

```diff
@@ -1,83 +1,31 @@
 import { Box, Divider, List, ListItem, ListItemText, Typography } from '@mui/material'
 import { observer } from 'mobx-react-lite'
 import Link from 'next/link'
-import { useState, MouseEvent } from 'react'
+import { MouseEvent, useEffect, useState } from 'react'
 
-import { DocumentContent, SourceField } from '../../../../../Types'
-import { flatten, formatDateTime, getLanguageName, humanFileSize, shortenName } from '../../../../../utils/utils'
+import { SourceField } from '../../../../../Types'
 import { useSharedStore } from '../../../../SharedStoreProvider'
 import { LinkMenu } from '../../../LinkMenu'
 
 import { useStyles } from './Meta.styles'
 
-const tableFields: Partial<
-    Record<
-        SourceField,
-        {
-            label: string
-            searchKey?: SourceField
-            visible?: (content?: Partial<DocumentContent>) => boolean
-            format?: (term?: any) => string
-            searchTerm?: (term: any) => string
-        }
-    >
-> = {
-    filename: {
-        label: 'Filename',
-    },
-    path: {
-        label: 'Path',
-        searchKey: 'path-parts',
-    },
-    filetype: {
-        label: 'Type',
-        visible: (content) => !!content?.filetype,
-    },
-    md5: {
-        label: 'MD5',
-        visible: (content) => content?.filetype !== 'folder' && !!content?.md5,
-    },
-    sha1: {
-        label: 'SHA1',
-        visible: (content) => content?.filetype !== 'folder' && !!content?.sha1,
-    },
-    lang: {
-        label: 'Language',
-        format: getLanguageName,
-        visible: (content) => !!content?.lang,
-    },
-    date: {
-        label: 'Modified',
-        format: formatDateTime,
-        visible: (content) => !!content?.date,
-    },
-    'date-created': {
-        label: 'Created',
-        format: formatDateTime,
-        visible: (content) => !!content?.['date-created'],
-    },
-    pgp: {
-        label: 'PGP',
-        format: () => 'true',
-        searchTerm: () => 'true',
-        visible: (content) => !!content?.pgp,
-    },
-    'word-count': {
-        label: 'Word count',
-        searchTerm: (term) => term.toString(),
-        visible: (content) => !!content?.['word-count'],
-    },
-    size: {
-        label: 'Size',
-        format: humanFileSize,
-        searchTerm: (term) => term.toString(),
-        visible: (content) => !!content?.size,
-    },
-}
-
 export const Meta = observer(() => {
     const { classes } = useStyles()
-    const { data, collection, collectionBaseUrl } = useSharedStore().documentStore
+    const {
+        data,
+        collection,
+        collectionBaseUrl,
+        metaStore: { tableData, metaData },
+        documentSearchStore: {
+            query,
+            setActiveSearch,
+            metaSearchStore,
+        },
+    } = useSharedStore().documentStore
+
+    useEffect(() => {
+        setActiveSearch(metaSearchStore)
+    }, [setActiveSearch, metaSearchStore])
```

Comment:
use reaction in `documentSearchStore` constructor on `documentStore.subTab` change instead of these effect

Reply:
Same as above, active searches need to be set, in the component. Its either this, or we need specific id's for each tab, and that way we can create a reaction that listens to that ID, and sets the active search. But for now, as far as I've seen, tab and subtab are index based, which can change from document to document, where index 0 is PDF search, and for others index 0 might be text search